pkg/cmd/infra/router/router.go

```diff
@@ -26,8 +27,24 @@ This command launches a router connected to your OpenShift master. The router li
 created by users and keeps a local router configuration up to date with those changes.`
 )
 
+// routerConfig represents a container for all other router config types.
+// Any global config belongs on this level.
+type routerConfig struct {
+	//Config is global flags that are passed to all client commands that need to connect to master
+	Config *clientcmd.Config
+
+	//Type is the type of plugin being created
+	Type string
+
+	//TemplateRouterConfig holds config items for template router plugins
+	TemplateRouterConfig *templateRouterConfig
```

Comment:
You can leave the `newEmptyConfig` method, it doesn't hurt anything, but yes, they can assume zero values since they're always needed for the cmd flag parsing.